### DISCUSSION TITLE
Data input formats for univariate and multivariate TS classification

### DISCUSSION THREAD

**Discussion by monasound** [alan-turing-institute/sktime#4495] [Q&A] [answered]
Thank you for the Sktime classifications package. 

I am doing both univariate and multivariate TS classifications. 

My question is about the data format. I have created data format for both univariate and multivariate. Both are producing results.
But I want to make sure that whether I am doing it correctly or not. Since I am completely new to Python, hence the doubt. I appreciate your reply on this question.

Below is the panel dataframe. Y is the binary dependent variable.

```
	 	           Var1	        Var2               Y
Instance	 Time	 	 
2200271	           0	5.527678	0.794594           1
 	           1	6.418026	0.697625           1
 	           2 	6.459351	0.693111           1
 	           3	8.967987	0.464545           1
 	           4	6.454385	0.691827           1

 	...	...	 
50000576	   5	5.667014	0.77729            0
 	           6	5.268255	0.822749           0
 	           7	5.176998	0.829841           0
 	           8	5.935367	0.751053           0
 	           9	5.091187	0.839757           0

```

**Univariate classification:** 
Input to the sktime TS is in panel (pandas dataframe). 
Instance =230, Time=10, variables=1 (taken only Var1)

**Output**
X = df.to_numpy().reshape(230, 10)

**Multivariate classification:** 
Input to the sktime TS is in panel (pandas dataframe). 
Instance =230, Time=10, Variables=2 (taken both Var1 & Var2 )

**Output** 
X = df.to_numpy().reshape(230,2,10)

I guess this is wrong for multivariate. Could you please let me know

**Comment by fkiraly** [comment, marked as answer]
This looks correct in both cases.

* as `numpy`, univariate could also be `(230, 1, 10)`.
* you can also just directly pass the `df` without the `Y` column, that would also work.
* In all cases, `y` should be a vector of length `230`.

**Reply by monasound** [reply]
Thank you so much for your reply.

**Comment by ABJ66** [comment]
Hi monasound/fkiraly,

Even I had a query on how to prepare the input data for multivariate ts classification problem. Discussion # 4458. Does it always have to be panel data/3d input data for the model, instances*features*timestamps? Because I have tried an iteration on 2d data (as per my use case it's to detect if a particular set of features using sensor readings can define an anomalous behavior or not) and it still worked (earlier iterations used panel/3d where I used a bunch of serially ordered timestamps as an instance). 

I don't see a requirement for grouping timestamps into instances though, but I could be mistaken.

Would be great to hear from the maintainers/user community on their thoughts.

Thanks.

**Reply by fkiraly** [reply]
@abj66, yes, one of the different formats that classifiers accept is numpy 2D. That is interpreted as (instances, timestamps).

However, it is important to be careful here, since transformers interpret numpy 2D as a single time series, of (timestamps, features).

For this reason, use of numpy 2D in time series classification is discouraged, but it is supported for convenience and due to legacy conventions.
